examples/error-handling/error_handling.bal

```diff
@@ -1,36 +1,40 @@
 import ballerina/io;
 
+// Define a record to represent error details.
 type AccountNotFoundErrorData record {
-    string accountID;
+    int accountID;
 };
 
-// Extended error types can be declared from `error` type. Thus `error` can be extended for single level.
+// User defined `error` types can be introduced based on the `error` basic type.
+// The reason type descriptor (a subtype of `string`) and optionally a `detail`
+// type descriptor (a subtype of `record {}` or `map<anydata>`) need to be specified.
 type AccountNotFoundError error<string, AccountNotFoundErrorData>;
 
-// Note that type `error` is a built-in reference type.
-function getAccountBalance(int accountID) returns (int|AccountNotFoundError) {
-    // Create an instance of the error record and return it.
-    // The logic used here is a sample to demonstrate the concept of error handling.
-    if (accountID < 100) {
-        AccountNotFoundError accountNotFoundError = error("Account with ID: "
-                                        + accountID + " is not found",
-                                        { accountID: <string>accountID });
+function getAccountBalance(int accountID) returns int|AccountNotFoundError {
+    // Return an error if the `accountID` is less than `0`.
+    if (accountID < 0) {
+        string errorReason = "Account Not Found";
+        AccountNotFoundErrorData errorDetail = {
+            accountID: accountID
+        };
+        AccountNotFoundError accountNotFoundError =
+                                            error(errorReason, errorDetail);
         return accountNotFoundError;
-    } else {
-        return 600;
     }
+    // Return a value if the `accountID` is greater than `0`.
+    return 600;
 }
 
 public function main() {
-    // As a best practice, check whether an error occurrs.
-    var r = getAccountBalance(24);
-
-    // Type-guard check (`is` check) for variable type checking.
-    if (r is int) {
-        io:println(r);
+    var result = getAccountBalance(-1);
+    // If `result` is an `int`, print the value.
+    if (result is int) {
+        io:println("Account Balance: ", result);
+    // If an error is returned, print the reason and the account ID from the detail map.
+    // The `.reason()` and `.detail()` built-in methods can be called on `error` typed
```

Comment:
```suggestion
    // The '.reason()' and '.detail()' built-in methods can be called on 'error' type
```

Reply:
Yes, `typed` seems more suitable.

Reply:
Saying "typed' denotes that the error is typed by the user. Also, to refer to a variable of the type "error" it is incorrect to say "error-typed variable". 

To be clear, shall we say: "variables of type`error'?